src/mesh/serial_mesh.C

```diff
@@ -1248,10 +1248,9 @@ void SerialMesh::stitching_helper (SerialMesh* other_mesh,
             {
               const std::vector<boundary_id_type>& ids =
                 other_mesh->get_boundary_info().boundary_ids(other_elem->get_node(n));
-              if (!ids.empty())
-                {
-                  this->get_boundary_info().add_node(this_elem->get_node(n), ids);
-                }
+              this->get_boundary_info().add_node(this_elem->get_node(n),
+                                                 std::set<boundary_id_type>(ids.begin(),
+                                                                            ids.end()));
```

Comment:
I think the answer is to deprecate more stuff as we go. `BoundaryInfo::boundary_ids(Node*)` could be deprecated in favor of a version which returns a `std::set` for instance.  But I figured enough had been done already in the current PR...